rocAL/include/loaders/audio/audio_loader.h

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright (c) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#pragma once
+
+#include <string>
+#include <thread>
+#include <vector>
+
+#include "loaders/audio/audio_read_and_decode.h"
+#include "loaders/circular_buffer.h"
+#include "pipeline/commons.h"
+#include "meta_data/meta_data_reader.h"
+
+#ifdef ROCAL_AUDIO
+
+// AudioLoader runs an internal thread for loading and decoding of audios asynchronously
+// It uses a circular buffer to store decoded audios for the user
+class AudioLoader : public LoaderModule {
+   public:
+    explicit AudioLoader(void* dev_resources);
+    ~AudioLoader() override;
+    LoaderModuleStatus load_next() override;
```

Comment:
Please be consistent with naming for functions in a class. They all need to be PascalCase. Like you used in constructor/destructor

Reply:
The functions present in this file are overriden functions from base class, inorder to change these to PascalCase we would need to change base class and all the other inherited classes. Hence retaining this for now, will make note of this and do this in cleanup PR